File: src/main/java/com/redbox/domain/user/dto/ReceptionResponse.java
package com.redbox.domain.user.dto;

import com.redbox.domain.donation.entity.DonationGroup;
import lombok.Getter;

import java.time.LocalDate;

@Getter
public class ReceptionResponse {

    private final String donorName;
    private final int donationAmount;
    private final LocalDate donationDate;
    private final String donationMessage;

    public ReceptionResponse(DonationGroup donationGroup, String donorName) {
        this.donorName = donorName;
        this.donationAmount = donationGroup.getDonationAmount();
        this.donationDate = donationGroup.getDonationDate();
        this.donationMessage = donationGroup.getDonationMessage();
    }
}

File: src/main/java/com/redbox/domain/user/dto/ReceptionResponse.kt
package com.redbox.domain.user.dto

import com.redbox.domain.donation.entity.DonationGroup
import java.time.LocalDate

data class ReceptionResponse(
    val donorName: String,
    val donationAmount: Int,
    val donationDate: LocalDate,
    val donationMessage: String
) {
    constructor(donationGroup: DonationGroup, donorName: String) : this(
        donorName = donorName,
        donationAmount = donationGroup.donationAmount,
        donationDate = donationGroup.donationDate,
        donationMessage = donationGroup.donationMessage
    )
}